### PR TITLE
ss-28: * validation sign-up

### PR DESCRIPTION
**Pull request by ValeriiArt**
added a minimum of 3 and a maximum of 30 characters to password validation

### CODE REVIEW

File: frontend/src/bundles/auth/components/sign-up-form/constants/constants.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { type UserSignUpRequestDto } from '~/bundles/users/users';
 const DEFAULT_SIGN_UP_PAYLOAD: UserSignUpRequestDto = {
```
Comment (Evcat6) [inline frontend/src/bundles/auth/components/sign-up-form/constants/constants.ts:3]: I think you should create custom type/interface for this case

File: shared/src/bundles/users/types/user-sign-up-request-dto.type.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 type UserSignUpRequestDto = {
     email: string;
     password: string;
+    repeatPassword: string;
```
Comment (Evcat6) [inline shared/src/bundles/users/types/user-sign-up-request-dto.type.ts:4]: I think we don't need to change this type

File: backend/src/bundles/users/user.service.ts [hunk 3]
```diff
@@ -18,7 +16,7 @@ class UserService implements IService {
     }
 
     public async find(
```
Comment (Evcat6) [inline backend/src/bundles/users/user.service.ts:18]: I think you should create in user.repository.ts few methods to find user

Something like this:

` public async find(data: object): Promise<UserEntity | undefined> {
        const user = await this.userModel
            .query()
            .select()
            .where(data)
            .first();
        if (!user) {
            return undefined;
        }
        return UserEntity.initialize(user);
    }
`

and extend it in this way

`public async findByEmail(email: string): Promise<UserEntity | undefined> {
      return await this.find({ email });
 }`

**Review by Evcat6 — CHANGES_REQUESTED** [review]

**Review by Evcat6 — APPROVED** [review]

File: backend/src/bundles/auth/auth.controller.ts
Comment (nikita-remeslov) [inline backend/src/bundles/auth/auth.controller.ts:1]: you should update route documentation

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
Conflicts. 
I general looks good

File: frontend/src/bundles/auth/components/sign-up-form/constants/constants.ts (hunk 1 above)
Comment (nikita-remeslov) [reply frontend/src/bundles/auth/components/sign-up-form/constants/constants.ts:3]: Why do you need this constant?

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**